diff --git a/services/HyprlandBackend.qml b/services/HyprlandBackend.qml
--- a/services/HyprlandBackend.qml
+++ b/services/HyprlandBackend.qml
@@ -14,5 +14,5 @@ Scope {
     property var focusedMonitor: Hyprland.focusedMonitor
 
     function switchWorkspaceRelative(direction) {
-        Hyprland.dispatch(`hl.dsp.focus({workspace = "r${direction === "next" ? "+1" : "-1"}"})`);
+        Hyprland.dispatch(`workspace r${direction === "next" ? "+1" : "-1"}`);
     }
@@ -28,14 +28,14 @@ Scope {
     }
 
     function focusWindow(id) {
-        Hyprland.dispatch(`hl.dsp.focus({ window = "address:${id}" })`);
+        Hyprland.dispatch(`focuswindow address:${id}`);
     }
     function closeWindow(id) {
-        Hyprland.dispatch(`hl.dsp.window.close({ window = "address:${id}" })`);
+        Hyprland.dispatch(`closewindow address:${id}`);
     }
     function switchWorkspace(id) {
-        Hyprland.dispatch(`hl.dsp.focus({ workspace = ${id} })`);
+        Hyprland.dispatch(`workspace ${id}`);
     }
     function moveWindowToWorkspace(id, wsId) {
-        Hyprland.dispatch(`hl.dsp.window.move({ workspace = ${wsId}, follow = false, window = "address:${id}" })`);
+        Hyprland.dispatch(`movetoworkspacesilent ${wsId},address:${id}`);
     }
